src/main/java/com/github/manolo8/darkbot/gui/tree/editors/ImagePicker.java

```diff
@@ -0,0 +1,75 @@
+package com.github.manolo8.darkbot.gui.tree.editors;
+
+import com.github.manolo8.darkbot.config.ImageWrapper;
+import com.github.manolo8.darkbot.gui.AdvancedConfig;
+import eu.darkbot.api.config.util.OptionEditor;
+import com.github.manolo8.darkbot.gui.utils.Strings;
+import eu.darkbot.api.config.ConfigSetting;
+
+import javax.swing.*;
+import javax.swing.filechooser.FileNameExtensionFilter;
+import java.awt.*;
+import java.io.File;
+
+public class ImagePicker extends JButton implements OptionEditor<ImageWrapper> {
+    private JFileChooser fc;
+    private ImageWrapper image;
+
+    public ImagePicker(){
+        addActionListener(e -> {
+            if (fc == null) {
+                String path = image == null ? "." : image.getPath();
+                fc = new JFileChooser(new File(path)) {
+                    @Override
+                    protected JDialog createDialog(Component parent) throws HeadlessException {
+                        JDialog dialog = super.createDialog(parent);
+                        dialog.setAlwaysOnTop(true);
+                        return dialog;
+                    }
+                };
+                FileNameExtensionFilter filter = new FileNameExtensionFilter(
+                        "Images Files", "png", "jpeg", "jpg", "gif");
+                fc.setFileFilter(filter);
+                fc.setAcceptAllFileFilterUsed(false);
+            }
+
+            if (fc.showOpenDialog(null) == JFileChooser.APPROVE_OPTION) {
+                image.setPath(fc.getSelectedFile().getAbsolutePath());
+                setEditing(image);
+            }
+        });
+
+    }
+    @Override
+    public void setText(String text) {
+        if(text != null && text.contains(File.separator) && text.length() > 30){
+            super.setText(".."+text.substring(text.lastIndexOf(File.separator, text.length()-30)));
```

Comment:
lastIndexOf can return -1
lastIndexOf will also try to cut at the *last* file separator, instead of the *first* file separator (starting from length - 30)

The idea is to find the closest file separator to len-30 and cut there

Also the formatting, this methid is missing many spaces